File: boj/[15565] 귀여운 라이언/15565.py
import sys
from collections import defaultdict

N, K = map(int, sys.stdin.readline().split())
dolls = [int(x) for x in sys.stdin.readline().split()]

start, end = 0, 0
count = defaultdict(int)
count[dolls[end]] += 1
answer = 1000001
while end < N:
    if count[1] == K:
        answer = min(answer, end - start + 1)
        count[dolls[start]] -= 1
        start += 1
    else:
        end += 1
        if end >= N: break
        count[dolls[end]] += 1
if answer == 1000001: answer = -1
print(answer)

File: boj/[15565] 귀여운 라이언/15565.kt
import kotlin.math.min

fun main(args: Array<String>) {
    val (N, K) = readln().split(" ").map { it.toInt() }
    val dolls = readln().split(" ").map { it.toInt() }

    var start = 0
    var end = 0
    var answer = Int.MAX_VALUE
    val count = mutableMapOf<Int, Int>().withDefault { 0 }
    count[dolls[end]] = 1

    while (end < N) {
        if (count[1] == K) {
            answer = min(answer, end - start + 1)
            count[dolls[start]] = count.getValue(dolls[start]).minus(1)
            start++
        } else {
            end++
            if (end < N) {
                count[dolls[end]] = count.getValue(dolls[end]).plus(1)
            }
        }
    }

    if (answer == Int.MAX_VALUE) {
        answer = -1
    }
    println(answer)
}
